### DISCUSSION TITLE
secure api req

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#3629] [Q&A]
i am facing an errror if any of registered user inspect the request headers and use any application like postman add added all the the headers and authorization in there and post bulk request.how to prevent this?like any method in pocketbase to limit the req per minite or use any pb hooks two decrypt token than send the req to server?or use any api key

**Comment by ganigeorgiev** [comment]
There is no builtin requests rate limiter at the moment if that's what you are looking (_it may be added eventually in the future as part of https://github.com/pocketbase/pocketbase/issues/1232 but remains a low priority_).

If you want more advanced network controls like rate limiting, IP whitelist/blacklist, headers rewrite, etc, the general recommendation for now is to put PocketBase behind a reverse-proxy like nginx, apache, caddy and make the configuration from there (for example, check the official [nginx docs](https://docs.nginx.com/nginx/admin-guide/security-controls/controlling-access-proxied-http/)).

A sample PocketBase nginx config can be found in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

**Reply by breakingbad45** [reply]
can we do like something that in every requst i pass something encrypted text than in on onbeforecreate req hook will decrypt the text and if match than proceed the request else reject?
